### DISCUSSION TITLE
Nuran

### DISCUSSION THREAD

**Discussion by Ulasaydo** [community/community#29996] [Other Feature Feedback, Questions, & Ideas]

**Comment by radar-blip** [comment]
Maybe this will help but I'm not for sure  https://support.github.com/request

Down near the bottom 'Restore deleted data'

**Comment by airtower-luna** [comment]
Was the deleted branch one on GitHub, or in your local repository?

Does it still exist on the other side? If yes, you can retrieve it from there.

Otherwise if it was on GitHub, ask support as @linuxlawson said. If it was local check [Git Internals - Maintenance and Data Recovery](https://git-scm.com/book/en/v2/Git-Internals-Maintenance-and-Data-Recovery).

**Comment by srj101** [comment]
This might help :)

## Pre-requisite: You have to know your last commit message from your deleted branch.

`git reflog`
# Search for message in the list
`a901eda HEAD@{18}: commit: <last commit message>`

# Now you have two options, either checkout revision or HEAD
`git checkout a901eda`
# Or
`git checkout HEAD@{18}`

# Create branch
`git branch recovered-branch`

# You may want to push that back to remote
`git push origin recovered-branch:recovered-branch`

## If you can't remember the last commit message: 
You might check this [Restore Deleted Branch](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/deleting-and-restoring-branches-in-a-pull-request)
